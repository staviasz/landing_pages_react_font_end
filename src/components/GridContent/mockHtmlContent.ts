export const htmlContent = {
  sectionId: 'id',
  title: 'title testing',
  html: `<p>Lorem ipsum dolor sit amet. In adipisci exercitationem sed iure sapiente <strong>Qui magnam ex accusamus
          enim</strong> et aspernatur voluptatem sit dolorem cumque est incidunt officia. Ex voluptatem harum ea architecto
          alias <em>Qui unde in voluptatem dolor eos libero fuga aut esse velit</em>? Cum necessitatibus reiciendis eum voluptas
          tenetur <a href="https://www.loremipzum.com" target="_blank">Ut quaerat eos obcaecati quas</a> est optio consequatur
          id ullam nisi aut consequatur iusto. Ea mollitia adipisci est dolores dolorumSed aliquid. </p>
          <ul>
          <li>Est sint veniam qui asperiores neque! </li>
          <li>Qui laborum architecto cum omnis minima et nemo impedit et esse molestiae. </li>
          <li>Aut debitis placeat cum voluptatem illo. </li>
          </ul>
          <p>Est molestias dolore <strong>Ut labore qui unde veritatis et galisum corporis</strong> et ipsam exercitationem est
          quas obcaecati. Et harum reprehenderit <em>In voluptatibus sit quaerat neque in ullam maiores</em> aut enim sapiente?
          </p>
          <p>Aut laborum nulla qui quos dolores <strong>Est numquam id quidem modi</strong>. Eum animi officiis ex ratione impedit
          <a href="https://www.loremipzum.com" target="_blank">Ut impedit et dolorem provident non dolor quia ea dolorem
            eveniet</a>. Eos molestias explicabo <em>Non quia</em> aut nemo galisum rem expedita sunt? Ut aliquam unde ut
          numquam quamNon doloremque sit corrupti reprehenderit id culpa nihil quo natus incidunt hic unde labore. </p>`,
  background: false,
};
