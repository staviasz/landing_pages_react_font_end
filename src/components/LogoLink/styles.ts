import styled, { css } from 'styled-components';

export const LinkContent = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
    > img {
      max-height: 3rem;
    }
  `}
`;
