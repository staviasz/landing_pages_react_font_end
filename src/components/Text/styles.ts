import styled, { css } from 'styled-components';

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    text-align: left;
  `}
`;
