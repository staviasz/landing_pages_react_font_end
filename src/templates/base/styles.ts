import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 4.4rem;
    overflow-y: auto;

    @media ${theme.media.lteMedium} {
      padding-top: 0;
    }
  `}
`;
