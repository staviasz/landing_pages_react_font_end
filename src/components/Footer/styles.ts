import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Text } from '../Text/styles';

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding: ${theme.spacings.xsmall};
      @media ${theme.media.lteMedium} {
        max-width: 80%;
        margin-left: 18px;
      }
    }

    ${Text} {
      font-size: ${theme.font.sizes.small};
      text-align: center;
      margin: 0;
    }
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
