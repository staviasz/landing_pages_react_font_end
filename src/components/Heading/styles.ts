import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface TitleProps {
  colordark: string;
  size: string;
  uppercase: string;
}

interface TitleSize {
  [key: string]: (theme: Theme) => ReturnType<typeof css>;
}

const mediaFont = (theme: Theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleSize: TitleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xhuge: (theme) => css`
    ${mediaFont(theme)}
    font-size: ${theme.font.sizes.xhuge};
  `,
};

export const Title = styled.h1<TitleProps>`
  ${({ theme, colordark, size, uppercase }) => css`
    color: ${colordark === 'dark'
      ? theme.colors.primaryColor
      : theme.colors.white};
    ${titleSize[size](theme)}
    text-transform: ${uppercase};
    font-family: ${theme.font.family.secondary};
    font-style: italic;
  `}
`;
