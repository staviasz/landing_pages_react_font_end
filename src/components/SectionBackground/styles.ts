import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface ContainerProps {
  background: string;
}

const containerBackgroundActivate = (theme: Theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div<ContainerProps>`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background === 'true' && containerBackgroundActivate(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
