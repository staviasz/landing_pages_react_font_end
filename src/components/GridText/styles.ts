import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    ${Title} {
      position: relative;
      left: 5rem;
    }
    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
