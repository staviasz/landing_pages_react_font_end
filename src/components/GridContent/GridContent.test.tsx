import { screen } from '@testing-library/react';
import { GridContent } from '.';
import { renderTheme } from '../../styles/render-theme';

import { theme } from '../../styles/theme';
import { htmlContent } from './mockHtmlContent';

describe('<GridContent />', () => {
  it('should render grid content', () => {
    const { container } = renderTheme(<GridContent {...htmlContent} />);
    expect(container).toMatchSnapshot();
  });

  it('should render grid content', () => {
    const { container } = renderTheme(<GridContent {...htmlContent} />);
    expect(container).toMatchSnapshot();
  });

  it('should have title "light" if background are "dark"', () => {
    renderTheme(<GridContent {...htmlContent} />);
    const heading = screen.getByRole('heading', { name: htmlContent.title });
    expect(heading).toHaveStyleRule('color', theme.colors.primaryColor);
  });

  it('should have title "light" if background are "dark"', () => {
    renderTheme(<GridContent {...htmlContent} background={true} />);
    const heading = screen.getByRole('heading', { name: htmlContent.title });
    expect(heading).toHaveStyleRule('color', theme.colors.white);
  });
});
