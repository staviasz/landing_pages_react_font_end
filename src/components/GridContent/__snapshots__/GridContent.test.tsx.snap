// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> should render grid content 1`] = `
.c3 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
  font-family: 'Montserrat',sans-serif;
  font-style: italic;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  text-align: left;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
  width: 100%;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          title testing
        </h2>
        <div
          class="c4"
        >
          <p
            class="c5"
          >
            <p>
              Lorem ipsum dolor sit amet. In adipisci exercitationem sed iure sapiente 
              <strong>
                Qui magnam ex accusamus
          enim
              </strong>
               et aspernatur voluptatem sit dolorem cumque est incidunt officia. Ex voluptatem harum ea architecto
          alias 
              <em>
                Qui unde in voluptatem dolor eos libero fuga aut esse velit
              </em>
              ? Cum necessitatibus reiciendis eum voluptas
          tenetur 
              <a
                href="https://www.loremipzum.com"
                target="_blank"
              >
                Ut quaerat eos obcaecati quas
              </a>
               est optio consequatur
          id ullam nisi aut consequatur iusto. Ea mollitia adipisci est dolores dolorumSed aliquid. 
            </p>
            
          
            <ul>
              
          
              <li>
                Est sint veniam qui asperiores neque! 
              </li>
              
          
              <li>
                Qui laborum architecto cum omnis minima et nemo impedit et esse molestiae. 
              </li>
              
          
              <li>
                Aut debitis placeat cum voluptatem illo. 
              </li>
              
          
            </ul>
            
          
            <p>
              Est molestias dolore 
              <strong>
                Ut labore qui unde veritatis et galisum corporis
              </strong>
               et ipsam exercitationem est
          quas obcaecati. Et harum reprehenderit 
              <em>
                In voluptatibus sit quaerat neque in ullam maiores
              </em>
               aut enim sapiente?
          
            </p>
            
          
            <p>
              Aut laborum nulla qui quos dolores 
              <strong>
                Est numquam id quidem modi
              </strong>
              . Eum animi officiis ex ratione impedit
          
              <a
                href="https://www.loremipzum.com"
                target="_blank"
              >
                Ut impedit et dolorem provident non dolor quia ea dolorem
            eveniet
              </a>
              . Eos molestias explicabo 
              <em>
                Non quia
              </em>
               aut nemo galisum rem expedita sunt? Ut aliquam unde ut
          numquam quamNon doloremque sit corrupti reprehenderit id culpa nihil quo natus incidunt hic unde labore. 
            </p>
          </p>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`<GridContent /> should render grid content 2`] = `
.c3 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
  font-family: 'Montserrat',sans-serif;
  font-style: italic;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  text-align: left;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
  width: 100%;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          title testing
        </h2>
        <div
          class="c4"
        >
          <p
            class="c5"
          >
            <p>
              Lorem ipsum dolor sit amet. In adipisci exercitationem sed iure sapiente 
              <strong>
                Qui magnam ex accusamus
          enim
              </strong>
               et aspernatur voluptatem sit dolorem cumque est incidunt officia. Ex voluptatem harum ea architecto
          alias 
              <em>
                Qui unde in voluptatem dolor eos libero fuga aut esse velit
              </em>
              ? Cum necessitatibus reiciendis eum voluptas
          tenetur 
              <a
                href="https://www.loremipzum.com"
                target="_blank"
              >
                Ut quaerat eos obcaecati quas
              </a>
               est optio consequatur
          id ullam nisi aut consequatur iusto. Ea mollitia adipisci est dolores dolorumSed aliquid. 
            </p>
            
          
            <ul>
              
          
              <li>
                Est sint veniam qui asperiores neque! 
              </li>
              
          
              <li>
                Qui laborum architecto cum omnis minima et nemo impedit et esse molestiae. 
              </li>
              
          
              <li>
                Aut debitis placeat cum voluptatem illo. 
              </li>
              
          
            </ul>
            
          
            <p>
              Est molestias dolore 
              <strong>
                Ut labore qui unde veritatis et galisum corporis
              </strong>
               et ipsam exercitationem est
          quas obcaecati. Et harum reprehenderit 
              <em>
                In voluptatibus sit quaerat neque in ullam maiores
              </em>
               aut enim sapiente?
          
            </p>
            
          
            <p>
              Aut laborum nulla qui quos dolores 
              <strong>
                Est numquam id quidem modi
              </strong>
              . Eum animi officiis ex ratione impedit
          
              <a
                href="https://www.loremipzum.com"
                target="_blank"
              >
                Ut impedit et dolorem provident non dolor quia ea dolorem
            eveniet
              </a>
              . Eos molestias explicabo 
              <em>
                Non quia
              </em>
               aut nemo galisum rem expedita sunt? Ut aliquam unde ut
          numquam quamNon doloremque sit corrupti reprehenderit id culpa nihil quo natus incidunt hic unde labore. 
            </p>
          </p>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
